tests/unit/controlgroup/core.js

```diff
@@ -74,4 +74,29 @@ QUnit.test( "selectmenu: controlgroupLabel", function( assert ) {
 	assert.hasClasses( label.find( "span" ), "ui-controlgroup-label-contents" );
 } );
 
+var assertSanatized = function( assert, initClasses, expectedClasses, message ) {
+	var selectmenu = $( "#select-sanatize" ).selectmenu({
+		classes: {
+			"ui-selectmenu-button-open": initClasses
+		}
+	}).selectmenu( "instance" );
+	var classes = {
+		"ui-selectmenu-button-open": "ui-corner-top"
+	};
+	var result = $.ui.controlgroup.prototype._resolveClassesValues( classes, selectmenu );
+	assert.deepEqual( result, {
+		"ui-selectmenu-button-open": expectedClasses + " ui-corner-top"
+	}, message );
+}
+
+QUnit.test( "_resolveClassesValues", function( assert ) {
+	assert.expect( 6 );
+	assertSanatized( assert, "bar ui-corner-bottom", "bar", "Single Corner Class Removed end" );
+	assertSanatized( assert, "ui-corner-bottom bar", "bar", "Single Corner Class Removed beginning" );
+	assertSanatized( assert, "bar ui-corner-bottom ui-corner-left", "bar", "Multiple Corner Class Removed end" );
+	assertSanatized( assert, "ui-corner-bottom ui-corner-left bar", "bar", "Multiple Corner Class Removed beginning" );
+	assertSanatized( assert, "bar ui-corner-bottom ui-corner-left foo", "bar foo", "Multiple Corner Class Removed Middle" );
+	assertSanatized( assert, "bar", "bar", "No corner Class" );
```

Comment:
Update the casing in the messages so only the first word is capitalized. Fix plurality of class in "multiple corner class".